src/cargo/ops/cargo_test.rs

```diff
@@ -144,6 +145,8 @@ fn run_doc_tests(options: &TestOptions,
             p.arg("--test").arg(lib)
              .arg("--crate-name").arg(&crate_name);
 
+            p.arg("-L").arg(&host_deps);
```

Comment:
It occurred to me that this line may need to be further down to put the host directory after the others? Does the ordering of the `-L` options mean this is buggy despite passing tests?